### ISSUE TITLE
Nested flexbox doesn't render correctly

### ISSUE THREAD

**Issue by zelitomas** [Kozea/WeasyPrint#1171]
Hi,

I have encountered a problem with nested flexbox that I can't figure out how to solve. If I use simple flexbox, everything works as expected, but when I try to nest them, strange things starts to happens. 

Here is the most simplified example I could come up with:

```
    #paper {
        width: 148mm;
        height: 105mm;
        
        display: flex;
        flex-direction: column;
    }
    #content {
        width: 100%;

        display: flex;
        flex-direction: column;
        flex: auto;
        justify-content: space-between;
    }
    #header {
        width: 100%;
        height: 2cm;
    }
```

``` html
<div id="paper">
    <div id="header" style="background-color:lightblue">Header part, should be full width, 2 cm height</div>

    <div id="content">
        <div style="background-color:orange" class="toppart">
            Middle part, should be 100% width, blank space should follow thanks to justify-items: between.
        </div>
        <div class="bottompart" style="background-color:yellow">
            Bottom part. Should be 100% width, blank space before.
        </div>                
    </div>
    
    
</div>
```
The expected result should look something like this:

![Snímek z 2020-07-23 01-50-50](https://user-images.githubusercontent.com/3105674/88240448-6d9edb00-cc87-11ea-9ae6-e475672240f5.png)

But the actual result looks like this:

![Snímek z 2020-07-23 01-55-08](https://user-images.githubusercontent.com/3105674/88240513-a2129700-cc87-11ea-8ec9-e2c2b09401e0.png)

I couldn't find similar bug, so I'm reporting it here, sorry if it is a duplicate. Is there anything I could do to help fix this issue?

**Comment by kingraphaii** [comment]
Having same issue here, version 51. Did you find a work around?

**Comment by zelitomas** [comment]
Unfortunately, I have not. And I'm not fluent enough in python to find the bug in the code, unfortunately.

**Comment by zelitomas** [comment]
Hi, Is there any way I could help resolving this issue?

**Comment by liZe** [comment]
> Hi, Is there any way I could help resolving this issue?
> 
> (adding more examples, etc)

I just need more time to fix this issue 🕐. The examples are really useful, thank you!

**Comment by sharicahill** [comment]
Hi, sorry for opening a new issue. I'll keep my comments in this one.

I've found a workaround for the width issue in case anyone else has the same problem. Not sure about the `justify-content: space-between;` issue with the blank space in the first example.

I think the issue with the widths is related to the percentages on the child elements. If you replace the percentages with fixed values, eg. `width: 100px;` it renders correctly. Using the flex property on the child elements also works, eg. `flex: 1;`

**Comment by liZe** [comment]
> I think the issue with the widths is related to the percentages on the child elements.

Then it’s probably related to #656 too.

> Hi, sorry for opening a new issue.

No problem, thanks for the bug report! It’s always useful to get different examples when there’s an issue.

**Comment by serbboy23** [comment]
Hello,
I'm also seeing this issue and have tested with version 54.0b1. The elements within my nested flexbox are not filling the width of my parent. 
As @sharicahill mentioned, the only workaround is to manually set the with of my elements .. however this is not ideal as some of my elements are dynamic is size.

**Comment by micycle1** [comment]
Another (simple) example, I believe:

### HTML+CSS
Expected layout
![image](https://user-images.githubusercontent.com/9304234/151804354-e488fd02-deb6-4e51-8aff-d781af3821a8.png)

### PDF
WeasyPrint layout
![image](https://user-images.githubusercontent.com/9304234/151804374-2bc41fd6-7ac3-4487-a2a2-9e304999cd8d.png)

```
@page {
    margin: 1em;
}

article {
    display: flex;
    flex-direction: column;
    max-height: 297mm;
    max-width: 210mm;
    align-content: center;
    margin: 3em 2em;
}

img {
    max-width: 100%;
    max-height: 100%;
    min-height: 0;
    min-width: 0;
    object-fit: contain;
}

article div {
    display: flex;
}
```

```
<html>

<head>
    <meta charset="utf-8">
    <link href="columns-test.css" rel="stylesheet">
    <title>test</title>
    <meta name="description" content="Report example">
</head>

<body>
    <article>
        <img src="figures/a.jpg">
        <img src="figures/a.jpg">
        <img src="figures/a.jpg">
        <div>
            <img src="figures/a.jpg">
            <img src="figures/a.jpg">
        </div>
    </article>
</body>

</html>
```


Adding the following doesn't help PDF output much either.

```
article div {
    width: 200px;
    height: 200px;
}
```

![image](https://user-images.githubusercontent.com/9304234/151805514-461f967e-6e1d-4b5f-b99a-d1ae45858f85.png)

**Comment by ankitpandeypagefreezer** [comment]
Please fix this issue.

**Comment by liZe** [comment]
> Please fix this issue.

Hi!

As maintainers, we’d also love to see this issue fixed. But…

Getting a proper flexbox rendering is long and complicated, and we currently don’t have enough free time to do this right now. The three main solutions you have to get this issue fixed is:
- To wait for the maintainers to spend their free time to work on this. As it’s long and complicated, it may not happen soon.
- To open a pull request.
- To pay someone (for example the maintainers) to fix this.

**Comment by vesters** [comment]
Hi liZe

Thank you very much for your reply. I could be really nice to get some flexbox support for this lovely piece of software.

What amount you estimate would be enough to get this done within a couple of monts maybe?

**Comment by liZe** [comment]
Hi @vesters,

We think that it would take between 1 and 2 weeks full time to get a solid implementation. It’s not that long, but it’s still too long to work seriously on this on our free time.

**Comment by vesters** [comment]
@liZe Thank you for getting back to me on this. 

It sounds reasonable - do you guys have a going hourly rate?

**Comment by liZe** [comment]
> It sounds reasonable - do you guys have a going hourly rate?

Let’s discuss about this by mail! You can reach us at contact@courtbouillon.org.

**Comment by onedapperterm** [comment]
At this date the issue is not yet fixed (V61.2), but a workaround is using tables, and yes, is a pain in the neck, but depending on the case it works fine :)
